src/components/pages/newRecipe/form.vue

```diff
@@ -0,0 +1,91 @@
+<template>
+    <div class="col-12 col-sm-9 col-md-6">
+        <div class="row justify-center">
+            <div class="col-12">
+                <q-input
+                    v-model="formItem.name"
+                    :label="inputLabels.name"
+                ></q-input>
+            </div>
+            <div class="col-12">
+                <q-input
+                    v-model="formItem.url"
+                    :label="!!errorItem['url'] ? errorItem['url'] : inputLabels.url"
+                    :error="!!errorItem['url']"
+                ></q-input>
+            </div>
+            <div class="col-12">
+                <q-btn
+                    class="full-width"
+                    color="primary"
+                    :disable="!formValid"
+                    :label="inputLabels.submit"
+                ></q-btn>
+            </div>
+        </div>
+        <div class="row justify-center content-center full-width-full-height">
+            <div class="col-12">
+                <h6>Review your addition</h6>
+                <div class="row" v-for="([key, value], index) in Object.entries(formItem)" :key="index">
+                    <div class="col-12">{{ key }}: {{ value }}</div>
+                </div>
+            </div>
+        </div>
+        {{ formValid }}
+        {{ errorItem }}
+    </div>
+</template>
+
+<script setup>
+import { computed, ref, watch } from 'vue';
+
+defineOptions({
+	name: "src-components-pages-newRecipe-form",
+});
+
+const inputLabels = {
+    name: 'Name',
+    url: 'URL to the recipe',
+    submit: 'Submit'
+};
+
+const formItem = ref({
+    name: '',
+    url: ''
+});
+
+const validateURL =  ({ val }) => {
+    if (!val) return true;
+    return /(https?:\/\/(?:www\.|(?!www))[a-zA-Z0-9][a-zA-Z0-9-]+[a-zA-Z0-9]\.[^\s]{2,}|www\.[a-zA-Z0-9][a-zA-Z0-9-]+[a-zA-Z0-9]\.[^\s]{2,}|https?:\/\/(?:www\.|(?!www))[a-zA-Z0-9]+\.[^\s]{2,}|www\.[a-zA-Z0-9]+\.[^\s]{2,})/i
+        .test(val) || 'Must be a valid URL (https)';
+};
+
+const formRules = {
+    name: [],
+    url: [ validateURL ]
+};
+
+const errorItem = ref({});
+const formValid = computed(() => {
+    return !!Object.keys(errorItem.value).length;
+});
+
+watch(formItem.value, (newValue, oldValue) => {
```

Comment:
Oooh, what is `watch()`? Interested to learn more about that.

Reply:
The documentation for Vue is pretty excellent - https://vuejs.org/guide/essentials/watchers

watchers are super helpful/necessary depending on your implementations. watchEffect especially so, but it can also really mess you up if you aren't careful